gtsam/discrete/tests/testDiscreteValues.cpp

```diff
@@ -0,0 +1,76 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/*
+ * testDiscreteValues.cpp
+ *
+ *  @date Jan, 2022
+ *  @author Frank Dellaert
+ */
+
+#include <CppUnitLite/TestHarness.h>
+#include <gtsam/base/Testable.h>
+#include <gtsam/discrete/DiscreteValues.h>
+#include <gtsam/discrete/Signature.h>
+
+#include <boost/assign/std/map.hpp>
+using namespace boost::assign;
+
+using namespace std;
+using namespace gtsam;
+
+/* ************************************************************************* */
+// Check markdown representation with a value formatter.
+TEST(DiscreteValues, markdownWithValueFormatter) {
+  DiscreteValues values;
+  values[12] = 1;  // A
+  values[5] = 0;   // B
+  string expected =
+      "|Variable|value|\n"
+      "|:-:|:-:|\n"
+      "|B|-|\n"
+      "|A|One|\n";
+  auto keyFormatter = [](Key key) { return key == 12 ? "A" : "B"; };
+  DiscreteValues::Names names{{12, {"Zero", "One", "Two"}}, {5, {"-", "+"}}};
+  string actual = values.markdown(keyFormatter, names);
+  EXPECT(actual == expected);
+}
+
+/* ************************************************************************* */
+// Check html representation with a value formatter.
+TEST(DiscreteValues, htmlWithValueFormatter) {
+  DiscreteValues values;
+  values[12] = 1;  // A
+  values[5] = 0;   // B
+  string expected =
+      "<div>\n"
+      "<table class='DiscreteValues'>\n"
```

Comment:
Should be double quotes escaped

Reply:
I was wondering. I did some searching and [this](https://docs.microsoft.com/en-us/cpp/c-language/c-string-literals?view=msvc-170) says  "The single quotation mark (') can be represented without an escape sequence."

Reply:
Or are yuo saying html needs double quotes?

Reply:
I think single is fine: https://www.w3.org/TR/html4/intro/sgmltut.html#h-3.2.2